json-schema/Request-Context-v2.json

```diff
@@ -0,0 +1,267 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "$id": "https://github.com/tewhatuora/schemas/json-schema/Request-Context",
+  "description": "JSON schema for request context custom header schema for apps integrating with HNZ FHIR APIs",
+  "$comment": "This schema version 2.0 extends the original Request-Context.json public schema, adding new properties.  Properties from the original version remain but some have been deprecated due to the improved support for HPI references in this version",
+  "type": "object",
+  "properties": {
+    "userIdentifier": {
+      "type": "string",
+      "description": "The alphanumeric user id as known to the system they have signed into (which is consuming the FHIR API)",
+      "example": "LJRHODE"
+    },
+    "userRole": {
+      "type": "string",
+      "enum": [
+          "PAT",
+          "PROV"
+      ]
+    },
+    "hpiPractitioner": {
+      "type": "object",
+      "description": "User health practitioner identification as an HPI FHIR reference",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "Fixed to refer to FHIR Practitioner resource type",
+          "example": "Practitioner",
+          "pattern": "^Practitioner$"
+        },
+        "identifier": {
+          "type": "object",
+          "description": "These attributes constitute a FHIR Identifier for an HPI Practitioner resource",
+          "properties": {
+            "use": {
+              "type": "string",
+              "description": "Fixed to official",
+              "const": "official"
+            },
+            "system": {
+              "type": "string",
+              "description": "Url that identifies a HPI Practitioner resource",
+              "const": "https://standards.digital.health.nz/ns/hpi-person-id"
+            },
+            "value": {
+              "type": "string",
+              "description": "The practitioner's Common Person Number or CPN as defined by the HPI Practitioner Index) in the form NNXXXX where N is numeric and X is Alphabetic",
+              "pattern": "^[0-9]{2}[A-Z0-9]{4}$",
+              "example": "91ZABY"
+            }
+          },
+          "required": [ "use", "system", "value"]
+        },
+        "display": {
+          "type": "string",
+          "description": "The full name of the practitioner", 
+          "example": "Lyndi Jane Rhode"
+        }
+      },
+      "required": ["type","identifier","display"]
+    },          
+    "registrationAuthorityNumber": {
```

Comment:
propose this is removed. We have secondary identifier, why not use reuse that? See comment on Secondary Identifier.